x-pack/filebeat/input/http_endpoint/handler.go

```diff
@@ -53,13 +59,19 @@ func (h *httpHandler) sendResponse(w http.ResponseWriter, status int, message st
 	io.WriteString(w, message)
 }
 
-func (h *httpHandler) publishEvent(obj common.MapStr) {
+func (h *httpHandler) publishEvent(obj common.MapStr, headers common.MapStr) {
 	event := beat.Event{
 		Timestamp: time.Now().UTC(),
 		Fields: common.MapStr{
 			h.messageField: obj,
 		},
 	}
+	if h.preserveOriginalEvent {
+		event.PutValue("event.original", obj.String())
```

Comment:
This is reconstructing the JSON by marshaling it again. That's not really the true original bytes received in the request. To get the raw bytes you would want to change `httpReadJSON` to `Decode` into a [`json.RawMessage`](https://golang.org/pkg/encoding/json/#RawMessage) so that you can defer the unmarshalling until after you've saved a copy the raw JSON bytes.

Reply:
This one would be a bit tough, because this is after iterating over multiple objects in a single incoming request, we don't want to include the whole event.original in each of the objects, I might contact you to discuss this one a bit more